### PR TITLE
Refactored Party Booking defines.

### PR DESCRIPTION
**Pull request by Emistry**
Adjusted the name of defines to better suit their purpose.

[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[//]: # (Describe at length, the changes that this pull request makes.)

**Affected Branches:** master

[//]: # (Master? Slave?)

**Issues addressed:**
https://github.com/HerculesWS/Hercules/issues/1234  references: https://github.com/rathena/rathena/commit/ed7157c80b0da86aa81d3cfee5e9b9679938b4e2

[//]: # (Issue Tracker Number if any.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2231)

File: src/map/party.h [hunk 1]
```diff
@@ -27,8 +27,8 @@
 
 #include <stdarg.h>
 
-#define PARTY_BOOKING_JOBS 6
-#define PARTY_BOOKING_RESULTS 10
+#define MAX_PARTY_BOOKING_JOBS 6
```
Comment (4144) [inline src/map/party.h]: why you on this defines, can you add also guard header for changed defines?
for example ``#define MAX_PARTY_BOOKING_JOBS 6`` should be changed to
```
#ifndef MAX_PARTY_BOOKING_JOBS
#define MAX_PARTY_BOOKING_JOBS 6
#endif
```

same for ``MAX_PARTY_BOOKING_RESULTS``

File: src/map/party.c [hunk 2]
```diff
@@ -1314,7 +1314,7 @@ static void party_booking_register(struct map_session_data *sd, short level, sho
 	pb_ad->p_detail.level = level;
 	pb_ad->p_detail.mapid = mapid;
 
-	for(i=0;i<PARTY_BOOKING_JOBS;i++)
+	for(i=0;i<MAX_PARTY_BOOKING_JOBS;i++)
```
Comment (Kenpachi2k13) [inline src/map/party.c:1317]:
```suggestion
	for (i = 0; i < MAX_PARTY_BOOKING_JOBS; i++)
```

File: src/map/party.c [hunk 3]
```diff
@@ -1364,7 +1364,7 @@ static void party_booking_update(struct map_session_data *sd, short *job)
 
 	pb_ad->expiretime = (int)time(NULL);// Update time.
 
-	for(i=0;i<PARTY_BOOKING_JOBS;i++)
+	for(i=0;i<MAX_PARTY_BOOKING_JOBS;i++)
```
Comment (Kenpachi2k13) [inline src/map/party.c:1367]:
```suggestion
	for (i = 0; i < MAX_PARTY_BOOKING_JOBS; i++)
```

File: src/map/party.c [hunk 4]
```diff
@@ -1380,7 +1380,7 @@ static void party_recruit_search(struct map_session_data *sd, short level, short
 #ifdef PARTY_RECRUIT
 	struct party_booking_ad_info *pb_ad;
 	int count = 0;
-	struct party_booking_ad_info* result_list[PARTY_BOOKING_RESULTS];
+	struct party_booking_ad_info* result_list[MAX_PARTY_BOOKING_RESULTS];
```
Comment (Kenpachi2k13) [inline src/map/party.c:1383]:
```suggestion
	struct party_booking_ad_info *result_list[MAX_PARTY_BOOKING_RESULTS];
```

File: src/map/party.c [hunk 5]
```diff
@@ -1391,7 +1391,7 @@ static void party_recruit_search(struct map_session_data *sd, short level, short
 	{
 		if ((level && (pb_ad->p_detail.level < level-15 || pb_ad->p_detail.level > level)))
 			continue;
-		if (count >= PARTY_BOOKING_RESULTS){
+		if (count >= MAX_PARTY_BOOKING_RESULTS){
```
Comment (Kenpachi2k13) [inline src/map/party.c:1394]:
```suggestion
		if (count >= MAX_PARTY_BOOKING_RESULTS) {
```

File: src/map/party.c [hunk 6]
```diff
@@ -1413,7 +1413,7 @@ static void party_booking_search(struct map_session_data *sd, short level, short
 	struct party_booking_ad_info *pb_ad;
 	int i;
 	int count = 0;
-	struct party_booking_ad_info* result_list[PARTY_BOOKING_RESULTS];
+	struct party_booking_ad_info* result_list[MAX_PARTY_BOOKING_RESULTS];
```
Comment (Kenpachi2k13) [inline src/map/party.c:1416]:
```suggestion
	struct party_booking_ad_info *result_list[MAX_PARTY_BOOKING_RESULTS];
```

File: src/map/party.c [hunk 7]
```diff
@@ -1424,14 +1424,14 @@ static void party_booking_search(struct map_session_data *sd, short level, short
 	for( pb_ad = dbi_first(iter); dbi_exists(iter); pb_ad = dbi_next(iter) ) {
 		if (pb_ad->index < lastindex || (level && (pb_ad->p_detail.level < level-15 || pb_ad->p_detail.level > level)))
 			continue;
-		if (count >= PARTY_BOOKING_RESULTS){
+		if (count >= MAX_PARTY_BOOKING_RESULTS){
```
Comment (Kenpachi2k13) [inline src/map/party.c:1427]:
```suggestion
		if (count >= MAX_PARTY_BOOKING_RESULTS) {
```

File: src/map/party.c [hunk 8]
```diff
@@ -1424,14 +1424,14 @@ static void party_booking_search(struct map_session_data *sd, short level, short
...
 			more_result = true;
 			break;
 		}
 		if (mapid == 0 && job == -1)
 			result_list[count] = pb_ad;
 		else if (mapid == 0) {
-			for(i=0; i<PARTY_BOOKING_JOBS; i++)
+			for(i=0; i<MAX_PARTY_BOOKING_JOBS; i++)
```
Comment (Kenpachi2k13) [inline src/map/party.c:1434]:
```suggestion
			for (i = 0; i < MAX_PARTY_BOOKING_JOBS; i++)
```

**Comment by Kenpachi2k13** [comment]
*Resolved conflicts*

File: src/map/party.c [hunk 9]
```diff
@@ -1380,23 +1382,23 @@ static void party_recruit_search(struct map_session_data *sd, short level, short
...
+		if (count >= MAX_PARTY_BOOKING_RESULTS) {
 			more_result = true;
 			break;
 		}
 		result_list[count] = pb_ad;
-		if( result_list[count] )
+		if (result_list[count])
 		{
 			count++;
 		}
```
Comment (Kenpachi2k13) [inline src/map/party.c:1404]:
```suggestion
		if (result_list[count] != NULL)
			count++;
```

File: src/map/party.c [hunk 10]
```diff
@@ -1413,32 +1415,32 @@ static void party_booking_search(struct map_session_data *sd, short level, short
...
 	struct DBIterator *iter = db_iterator(party->booking_db);
 
 	nullpo_retv(sd);
 
 	memset(result_list, 0, sizeof(result_list));
 
-	for( pb_ad = dbi_first(iter); dbi_exists(iter); pb_ad = dbi_next(iter) ) {
-		if (pb_ad->index < lastindex || (level && (pb_ad->p_detail.level < level-15 || pb_ad->p_detail.level > level)))
+	for (pb_ad = dbi_first(iter); dbi_exists(iter); pb_ad = dbi_next(iter)) {
+		if (pb_ad->index < lastindex || (level && (pb_ad->p_detail.level < level - 15 || pb_ad->p_detail.level > level)))
```
Comment (Kenpachi2k13) [inline src/map/party.c:1427]:
```suggestion
		if (pb_ad->index < lastindex || (level != 0 && (pb_ad->p_detail.level < level - 15 || pb_ad->p_detail.level > level)))
```

File: src/map/party.c [hunk 11]
```diff
@@ -1413,32 +1415,32 @@ static void party_booking_search(struct map_session_data *sd, short level, short
...
-		} else if (job == -1){
+		} else if (job == -1) {
 			if (pb_ad->p_detail.mapid == mapid)
 				result_list[count] = pb_ad;
 		}
-		if( result_list[count] )
+		if (result_list[count])
 		{
 			count++;
 		}
```
Comment (Kenpachi2k13) [inline src/map/party.c:1446]:
```suggestion
		if (result_list[count] != NULL)
			count++;
```

File: src/map/clif.c [hunk 12]
```diff
@@ -13900,16 +13900,17 @@ static void clif_parse_PartyBookingRegisterReq(int fd, struct map_session_data *
 static void clif_parse_PartyBookingRegisterReq(int fd, struct map_session_data *sd)
 {
 #ifndef PARTY_RECRUIT
+	nullpo_retv(sd);
+
 	if (pc_istrading(sd) || pc_isvending(sd))
```
Comment (Kenpachi2k13) [inline src/map/clif.c:13905]:
```suggestion
	if (pc_istrading(sd) || pc_isvending(sd))
```

File: src/map/clif.c [hunk 13]
```diff
@@ -14038,12 +14039,14 @@ static void clif_parse_PartyBookingUpdateReq(int fd, struct map_session_data *sd
 static void clif_parse_PartyBookingUpdateReq(int fd, struct map_session_data *sd)
 {
 #ifndef PARTY_RECRUIT
+	nullpo_retv(sd);
+
 	if (pc_istrading(sd) || pc_isvending(sd))
```
Comment (Kenpachi2k13) [inline src/map/clif.c:14044]:
```suggestion
	if (pc_istrading(sd) || pc_isvending(sd))
```

**Comment by 4144** [comment]
and here merge commit too

**Comment by Emistry** [comment]
i have reset to first commit, then commit changes. kindly help re-review it in case i missed anything else.

File: src/map/clif.c [hunk 14]
```diff
@@ -14060,20 +14058,20 @@ static void clif_PartyBookingInsertNotify(struct map_session_data *sd, struct pa
...
 		return;
 
-	int i;
-	uint8 buf[38 + PARTY_BOOKING_JOBS * 2];
+	uint8 buf[38 + MAX_PARTY_BOOKING_JOBS * 2];
 
 	nullpo_retv(sd);
-	if(pb_ad == NULL) return;
+	if (pb_ad == NULL) return;
```
Comment (Kenpachi2k13) [inline src/map/clif.c:14064]:
```suggestion
	nullpo_retv(pb_ad);
```
And move nullpo checks to top.

File: src/map/clif.c [hunk 15]
```diff
@@ -14086,17 +14084,18 @@ static void clif_PartyBookingInsertNotify(struct map_session_data *sd, struct pa
...
 {
 #ifndef PARTY_RECRUIT
-	int i;
-	uint8 buf[6+PARTY_BOOKING_JOBS*2];
+	uint8 buf[6 + MAX_PARTY_BOOKING_JOBS * 2];
 
 	nullpo_retv(sd);
-	if(pb_ad == NULL) return;
+	if (pb_ad == NULL) 
+		return;
```
Comment (Kenpachi2k13) [inline src/map/clif.c:14091]:
```suggestion
	nullpo_retv(pb_ad);
```
And move nullpo checks to top.

File: src/map/clif.c (hunk 14 above)
Comment (Emistry) [reply src/map/clif.c:14064]: updated

File: src/map/clif.c (hunk 15 above)
Comment (Emistry) [reply src/map/clif.c]: updated

File: src/map/clif.c [hunk 16]
```diff
@@ -14267,24 +14265,25 @@ static void clif_parse_PartyBookingUpdateReq(int fd, struct map_session_data *sd
 /// 0809 <index>.L <char name>.24B <expire time>.L <level>.W <map id>.W { <job>.W }*6
 static void clif_PartyBookingInsertNotify(struct map_session_data *sd, struct party_booking_ad_info *pb_ad)
 {
+	nullpo_retv(sd);
+	if (pb_ad == NULL)
```
Comment (Kenpachi2k13) [inline src/map/clif.c:14269]: Why not `nullpo_retv(pb_ad);`?

File: src/map/clif.c [hunk 17]
```diff
@@ -14296,18 +14295,19 @@ static void clif_PartyBookingInsertNotify(struct map_session_data *sd, struct pa
 /// 080a <index>.L { <job>.W }*6
 static void clif_PartyBookingUpdateNotify(struct map_session_data *sd, struct party_booking_ad_info *pb_ad)
 {
+	nullpo_retv(sd);
+	if (pb_ad == NULL)
```
Comment (Kenpachi2k13) [inline src/map/clif.c:14299]: Why not `nullpo_retv(pb_ad);`?

Comment (Emistry) [reply src/map/clif.c:14299]: done

File: src/map/clif.c (hunk 16 above)
Comment (Emistry) [reply src/map/clif.c]: done

File: src/map/party.c [hunk 18]
```diff
@@ -1433,35 +1429,33 @@ static void party_booking_search(struct map_session_data *sd, short level, short
...
-		if (count >= PARTY_BOOKING_RESULTS){
+		if (count >= MAX_PARTY_BOOKING_RESULTS) {
 			more_result = true;
 			break;
 		}
 		if (mapid == 0 && job == -1)
 			result_list[count] = pb_ad;
 		else if (mapid == 0) {
-			for(i=0; i<PARTY_BOOKING_JOBS; i++)
+			for (i = 0; i < MAX_PARTY_BOOKING_JOBS; i++)
```
[the remark this answers is not in the document]
Comment (Emistry) [reply src/map/party.c]: done

File: src/map/party.c [hunk 19]
```diff
@@ -1377,17 +1375,18 @@ static void party_booking_update(struct map_session_data *sd, short *job)
...
 
-	if( pb_ad == NULL )
+	if (pb_ad == NULL)
 		return;
 
 	pb_ad->expiretime = (int)time(NULL);// Update time.
 
-	for(i=0;i<PARTY_BOOKING_JOBS;i++)
-		if(job[i] != 0xFF)
+	for (i = 0; i < MAX_PARTY_BOOKING_JOBS; i++)
```
[the remark this answers is not in the document]
Comment (Emistry) [reply src/map/party.c]: done

File: src/map/party.c [hunk 20]
```diff
@@ -1317,27 +1317,25 @@ static void party_booking_register(struct map_session_data *sd, short level, sho
...
 
-	memcpy(pb_ad->charname,sd->status.name,NAME_LENGTH);
+	memcpy(pb_ad->charname, sd->status.name, NAME_LENGTH);
 	pb_ad->expiretime = (int)time(NULL);
 	pb_ad->p_detail.level = level;
 	pb_ad->p_detail.mapid = mapid;
 
-	for(i=0;i<PARTY_BOOKING_JOBS;i++)
-		if(job[i] != 0xFF)
+	for (i = 0; i < MAX_PARTY_BOOKING_JOBS; i++)
```
[the remark this answers is not in the document]
Comment (Emistry) [reply src/map/party.c]: done

**Review by Kenpachi2k13 — APPROVED** [review]

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**